tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_PrimsAfterRollback.cpp

```diff
@@ -0,0 +1,108 @@
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Mesh.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "Evolution/DgSubcell/Tags/DidRollback.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/NewtonianEuler/PrimitiveFromConservative.hpp"
+#include "Evolution/Systems/NewtonianEuler/Subcell/PrimsAfterRollback.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+
+namespace {
+template <size_t Dim>
+void test(const gsl::not_null<std::mt19937*> gen,
+          const gsl::not_null<std::uniform_real_distribution<>*> dist,
+          const bool did_rollback) {
+  using MassDensityCons = NewtonianEuler::Tags::MassDensityCons;
+  using EnergyDensity = NewtonianEuler::Tags::EnergyDensity;
+  using MomentumDensity = NewtonianEuler::Tags::MomentumDensity<Dim>;
+
+  using MassDensity = NewtonianEuler::Tags::MassDensity<DataVector>;
+  using Velocity = NewtonianEuler::Tags::Velocity<DataVector, Dim>;
+  using SpecificInternalEnergy =
+      NewtonianEuler::Tags::SpecificInternalEnergy<DataVector>;
+  using Pressure = NewtonianEuler::Tags::Pressure<DataVector>;
+
+  using cons_tags = tmpl::list<MassDensityCons, MomentumDensity, EnergyDensity>;
+  using ConsVars = Variables<cons_tags>;
+  using prim_tags =
+      tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>;
+  using PrimVars = Variables<prim_tags>;
+
+  const Mesh<Dim> dg_mesh{5, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<Dim> subcell_mesh = evolution::dg::subcell::fd::mesh(dg_mesh);
+
+  auto cons_vars = make_with_random_values<ConsVars>(
+      gen, dist, subcell_mesh.number_of_grid_points());
+  PrimVars prim_vars{};
+
+  std::unique_ptr<EquationsOfState::EquationOfState<false, 1>> eos =
+      std::make_unique<EquationsOfState::PolytropicFluid<false>>(1.4,
+                                                                 5.0 / 3.0);
+
+  auto box = db::create<db::AddSimpleTags<
+      evolution::dg::subcell::Tags::DidRollback, ::Tags::Variables<cons_tags>,
+      ::Tags::Variables<prim_tags>, ::domain::Tags::Mesh<Dim>,
+      evolution::dg::subcell::Tags::Mesh<Dim>,
+      hydro::Tags::EquationOfState<
+          std::unique_ptr<EquationsOfState::EquationOfState<false, 1>>>>>(
+      did_rollback, cons_vars, prim_vars, dg_mesh, subcell_mesh,
+      std::move(eos));
+
+  db::mutate_apply<NewtonianEuler::subcell::PrimsAfterRollback<Dim>>(
+      make_not_null(&box));
+
+  if (did_rollback) {
+    REQUIRE(
```

Comment:
For my understanding: why is this a `REQUIRE`? It's one of the interface promises of the function, why not check with a `CHECK` as usual?

Reply:
because if the size is wrong the values are certainly wrong. I'm not super consistent with these sorts of checks, but I sometimes find it easier to have a clean end to a test rather than have it spit out a bunch of numbers that are wrong more or less as a symptom

Reply:
Okay, that's a fair argument ... thanks for explaining! (no need to reply in the other PR)